src/fal/run_scripts.py

```diff
@@ -46,48 +50,98 @@ class ContextConfig:
 class Context:
     current_model: Union[CurrentModel, None]
     config: ContextConfig
+    unique_id: UUID
 
 
 def run_scripts(list: List[FalScript], project: FalProject):
 
     print_run_info(list)
 
     faldbt = project._faldbt
-    for script in list:
-        model = script.model
-        meta = model.meta
-        _del_key(meta, project.keyword)
-
-        tests = _process_tests(model.tests)
-
-        current_model = CurrentModel(
-            name=model.name,
-            status=model.status,
-            columns=model.columns,
-            tests=tests,
-            meta=meta,
-        )
-
-        context_config = ContextConfig(_get_target_path(faldbt._config))
-        context = Context(current_model=current_model, config=context_config)
-
-        logger.info("Running script {} for model {}", script.path, model.name)
 
-        script.exec(context, faldbt)
-
-
-def run_global_scripts(list: List[FalScript], project: FalProject):
+    done_scripts: Dict[UUID, bool] = {}
+    pool = ThreadPool(faldbt.threads)
+    logger.info("Concurrency: {} threads", faldbt.threads)
+    try:
 
-    print_run_info(list)
+        for script in list:
+            context = _build_script_context(script, project)
+
+            done_scripts[context.unique_id] = False
+
+            if _is_global_script(script):
+                logger.debug("Running GLOBAL script {}", script.path)
+            else:
+                logger.debug(
+                    "Running script {} for model {}",
+                    script.path,
+                    script.model.name,
+                )
+
+            def callback(tpl: Tuple[FalScript, Context, Exception]):
+                script, context, err = tpl
+                if _is_global_script(script):
+                    logger.debug("Finished GLOBAL script {}", script.path)
+                else:
+                    logger.debug(
+                        "Finished script {} for model {}",
+                        script.path,
+                        script.model.name,
+                    )
+                done_scripts[context.unique_id] = True
+                if err is not None:
+                    # TODO: what to do?
```

Comment:
Once all the scripts are done, we should call `sys.exit` with non-zero exit code.

https://docs.python.org/3.8/library/sys.html#sys.exit

Reply:
this case is for when a script had an error